### PR TITLE
More code coverage

### PR DESCRIPTION
**Pull request by ITaluone**
This ref. #1823 

I don't know if these tests are important, but in case they are.. 

## IMPORTANT 

* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).

### CODE REVIEW

File: Src/FluentAssertions/Common/Guard.cs [hunk 1]
```diff
@@ -8,10 +8,7 @@ internal static class Guard
     {
         public static void ThrowIfArgumentIsNull<T>([ValidatedNotNull] T obj, string paramName)
         {
-            if (obj is null)
-            {
-                throw new ArgumentNullException(paramName);
-            }
+            ThrowIfArgumentIsNull(obj, paramName, string.Empty);
```
Comment (jnyrup) [inline Src/FluentAssertions/Common/Guard.cs:11]: Are you aware that this changes behavior?
I think the former, i.e. keeping it as it is, is better.
image

Also applies to the change to `ThrowIfArgumentIsNullOrEmpty`.

**Review by jnyrup — CHANGES_REQUESTED** [review]
Besides the single comment I'm good with this.

**Comment by ITaluone** [comment]
I don't know how to arrange a test for `ThrowIfArgumentIsOutOfRange`.. Because `T` is an `enum`

**Comment by eNeRGy164** [comment]
> I don't know how to arrange a test for `ThrowIfArgumentIsOutOfRange`.. Because `T` is an `enum`

Cast any enum to a value that's not listed. For example `(CSharpAccessModifier)1000`

**Comment by ITaluone** [comment]
ahh.. thank you :)

**Comment by dennisdoomen** [comment]
To be honest, I see no value in those extra tests. This is internal code that is already covered by all other code and only exists because of that other code. In fact, `ThrowIfArgumentContainsNull` can even be removed since it is dead code anyway.

**Comment by IT-VBFK** [comment]
Ok, removing dead code counts as "improve code coverage" IMO :)

**Comment by dennisdoomen** [comment]
There's an overload.

**Comment by IT-VBFK** [comment]
How I can see the code coverage on coveralls for the latest CI build?

**Comment by jnyrup** [comment]
I've _just_ started the CI pipeline.
Once it's done in ~10min the Coveralls bot will post a comment in this PR with a summary of the results.
To view the individual files you'll have to login to Coveralls with your GitHub account.

**Comment by jnyrup** [comment]
And if you can't wait for the bot to post, here are the change for `Guard.cs`
https://coveralls.io/builds/47932197/source?filename=Src%2FFluentAssertions%2FCommon%2FGuard.cs

**Comment by jnyrup** [comment]
Instead of adding tests of `ThrowIfArgumentContainsNull<T>(IEnumerable<T> values, string paramName, string message)` we could delete it, as it is dead code.

According to https://coveralls.io/builds/47932179/source?filename=Src%2FFluentAssertions%2FCommon%2FGuard.cs#L58
all the other lines are already covered by other tests.

**Comment by dennisdoomen** [comment]
> Instead of adding tests of `ThrowIfArgumentContainsNull<T>(IEnumerable<T> values, string paramName, string message)` we could delete it, as it is dead code.

That's exactly what I said ;-)

However, I don't get why Coverall did not post an update here. @eNeRGy164 any idea why it sometimes works and sometimes doesn't?

**Comment by IT-VBFK** [comment]
> There's an overload.

Here is an example of what I have thought about to be able to remove `ThrowIfArgumentContainsNull<T>()`.

This applies to all four assertions..

```csharp

public MethodInfoSelector(IEnumerable<Type> types)
{
    types.ToList()
        .ForEach(type => Guard.ThrowIfArgumentIsNull(type, nameof(type)));
    Guard.ThrowIfArgumentIsNull(types, nameof(types));

    selectedMethods = types.SelectMany(t => t
        .GetMethods(BindingFlags.DeclaredOnly | BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic)
        .Where(method => !HasSpecialName(method)));
}

```

because removing the overload would be a breaking change IMO

**Comment by jnyrup** [comment]
It's non-breaking to remove any unused members of `Guard` since the class is `internal`.

So `public static void ThrowIfArgumentContainsNull<T>(IEnumerable<T> values, string paramName)` should be kept as it is used.

`public static void ThrowIfArgumentContainsNull<T>(IEnumerable<T> values, string paramName, string message)` can be removed as it unused and not public because `Guard` is `internal`.

**Comment by dennisdoomen** [comment]
Maybe you're using an IDE that doesn't allow you to find usages, but my IDE (Rider) clearly tells me that the overload of `ThrowIfArgumentContainsNull` that takes three parameters isn't used at all. 

image

**Comment by ITaluone** [comment]
ok, that was completely my fault 🙈 

I have written the tests and therefore it showes up "2 usages".. (the tests)

I should have checked that before writing the tests :D

**Comment by eNeRGy164** [comment]
> However, I don't get why Coverall did not post an update here. @eNeRGy164 any idea why it sometimes works and sometimes doesn't?

At first, there was no run CI initiated by default because it's @ITaluone's first contribution.
But why there was no post after the actual run, I also don't know. I can only guess that maybe the API between Coveralls and GitHub is overloaded or something like that. Can't see anything specific.

**Comment by dennisdoomen** [comment]
> I should have checked that before writing the tests :D

Those tests can be removed anyway.

**Comment by ITaluone** [comment]
ok, got it

**Review by jnyrup — APPROVED** [review]

**Pull request merged**